File: edentalbackend-main/DentalOffice.Entities/dbEntities/Payment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace DentalOffice.Entities.dbEntities
{
    public class Payment
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public decimal Amount { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public int TreatmentId { get; set; }
        public string PaymentIntent { get; set; }
        public string CardNumber { get; set; }

        public Treatment Treatment { get; set; }
        [NotMapped]
        public string DentistId { get; set; }
        [NotMapped]
        public string Client { get; set; }
        [NotMapped]
        public string TreatmentName { get; set; }
    }
}

File: eDental-flutter-main/lib/models/payment.dart
import 'package:dart_json_mapper/dart_json_mapper.dart';

import '../helpers/dateJsonFormatter.dart';

@jsonSerializable
class Payment {
  int? id;
  @JsonProperty(converter: CustomDateTimeConverter())
  DateTime date;
  double amount;
  int userId;
  int treatmentId;
  String? cardNumber;
  String? client;
  String? treatmentName;
  String? dentistName;
  String? paymentIntent;
  Payment(
      this.date, this.amount, this.userId, this.treatmentId, this.cardNumber,
      {this.client,
      this.treatmentName,
      this.id,
      this.dentistName,
      this.paymentIntent});
}
